fuel-core/src/executor.rs

```diff
@@ -1594,4 +1686,183 @@ mod tests {
             assert!(maybe_utxo.is_none());
         }
     }
+
+    /// Helper to build transactions and a message in it for some of the message tests
+    fn make_tx_and_message(rng: &mut StdRng, da_height: u64) -> (Transaction, CheckedDaMessage) {
+        let mut message = DaMessage {
+            sender: rng.gen(),
+            recipient: rng.gen(),
+            owner: rng.gen(),
+            nonce: rng.gen(),
+            amount: 1000,
+            data: vec![],
+            da_height,
+            fuel_block_spend: None,
+        };
+
+        let tx = TransactionBuilder::script(vec![], vec![])
+            .add_unsigned_message_input(
+                rng.gen(),
+                message.sender,
+                message.recipient,
+                message.nonce,
+                message.amount,
+                vec![],
+            )
+            .finalize();
+
+        if let Input::MessageSigned { owner, .. } = tx.inputs()[0] {
+            message.owner = owner;
+        } else {
+            unreachable!();
+        }
+
+        (tx, message.check())
+    }
```

Comment:
Personally, I like to put helper functions at the top of the `tests` module in its own `helpers` submodule, i.e.: `tests::helpers::make_tx_and_message`. This way:
- When reading the code the test suite isn't interrupted by helper fns
- The test code makes it clear you're calling a test function, i.e. `helpers::make_tx_and_message(..)`

Reply:
I like to keep code like this near the test cases that use it, so I'm keeping it here for now. I agree that more-generic functions should be in helpers module.